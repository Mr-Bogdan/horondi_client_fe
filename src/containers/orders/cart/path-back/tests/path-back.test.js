import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as reactRedux from 'react-redux';

import PathBack from '../path-back';

configure({ adapter: new Adapter() });

let wrapper;

function spyOnSelector(lightMode, language) {
  jest.spyOn(reactRedux, 'useSelector').mockImplementation((cb) =>
    cb({
      Theme: { lightMode },
      Language: { language }
    })
  );
}

describe('Path back component tests', () => {
  it('should match snapshot', () => {
    spyOnSelector(true, 0);

    wrapper = shallow(<PathBack />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should cover other branches', () => {
    spyOnSelector(false, 1);

    wrapper = shallow(<PathBack />);
  });
});
