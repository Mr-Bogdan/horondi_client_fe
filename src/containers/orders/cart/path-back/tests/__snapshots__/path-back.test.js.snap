// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page tests should match snapshot 1`] = `
<div
  className="makeStyles-whiteThemePath-1"
>
  <Link
    to="/"
  >
    Головна
  </Link>
   / 
  <Link
    to="/catalog/:category"
  >
    Каталог
  </Link>
   / 
  Ваш кошик
</div>
`;
